--- proj/src/src/pj_mutex.c	2011-10-25 19:52:57.459416400 +0200
+++ proj/patch/src/pj_mutex.c	2011-10-25 19:46:39.142376300 +0200
@@ -176,18 +176,18 @@
 /*                                                                      */
 /*      Release the PROJ.4 lock.                                        */
 /************************************************************************/
 
 void pj_release_lock()
 {
     if( mutex_lock == NULL )
         pj_init_lock();
-
-    ReleaseMutex( mutex_lock );
+	else
+		ReleaseMutex( mutex_lock );
 }
 
 /************************************************************************/
 /*                          pj_cleanup_lock()                           */
 /************************************************************************/
 void pj_cleanup_lock()
 {
     if( mutex_lock != NULL )
@@ -200,13 +200,13 @@
 /************************************************************************/
 /*                            pj_init_lock()                            */
 /************************************************************************/
 
 static void pj_init_lock()
 
 {
     if( mutex_lock == NULL )
-        mutex_lock = CreateMutex( NULL, TRUE, NULL );
+        mutex_lock = CreateMutex( NULL, FALSE, NULL );
 }
 
 #endif // def MUTEX_win32
 
