Index: dev/geos/src/headers/geos/operation/overlay/OverlayNodeFactory.h
===================================================================
--- dev/geos/src/headers/geos/operation/overlay/OverlayNodeFactory.h	(révision 3823)
+++ dev/geos/src/headers/geos/operation/overlay/OverlayNodeFactory.h	(copie de travail)
@@ -8,7 +8,7 @@
  *
  * This is free software; you can redistribute and/or modify it under
  * the terms of the GNU Lesser General Public Licence as published
- * by the Free Software Foundation. 
+ * by the Free Software Foundation.
  * See the COPYING file for more information.
  *
  **********************************************************************
@@ -46,6 +46,7 @@
  */
 class GEOS_DLL OverlayNodeFactory: public geomgraph::NodeFactory {
 public:
+	OverlayNodeFactory() {};
 	geomgraph::Node* createNode(const geom::Coordinate &coord) const;
 	static const geomgraph::NodeFactory &instance();
 };
