--- src/geom/GeometryFactory.cpp	2011-07-20 18:24:14.593392500 +0200
+++ patch/geom/GeometryFactory.cpp	2011-10-24 11:17:25.365221700 +0200
@@ -671,7 +671,9 @@
 GeometryFactory::createGeometry(const Geometry *g) const
 {
 	// could this be cached to make this more efficient? Or maybe it isn't enough overhead to bother
-	return g->clone();
+	Geometry* newGeom(g->clone());
+	newGeom->factory = this;
+	return newGeom;
 	//GeometryEditor *editor=new GeometryEditor(this);
 	//gfCoordinateOperation *coordOp = new gfCoordinateOperation();
 	//Geometry *ret=editor->edit(g, coordOp);
