--- src/operation/distance/DistanceOp.cpp	2014-02-11 16:53:31.366630500 +0100
+++ patch/operation/distance/DistanceOp.cpp	2014-03-28 18:07:50.645566865 +0100
@@ -390,8 +390,12 @@
 	 */
 	LineString::ConstVect lines0;
 	LineString::ConstVect lines1;
-	LinearComponentExtracter::getLines(*(geom[0]), lines0);
-	LinearComponentExtracter::getLines(*(geom[1]), lines1);
+
+	LinearComponentExtracter lce0(lines0);
+	geom[0]->apply_ro(&lce0);
+
+	LinearComponentExtracter lce1(lines1);
+	geom[1]->apply_ro(&lce1);
 
 #if GEOS_DEBUG
 	std::cerr << "LinearComponentExtracter found "
@@ -402,8 +406,12 @@
 
 	Point::ConstVect pts0;
 	Point::ConstVect pts1;
-	PointExtracter::getPoints(*(geom[0]), pts0);
-	PointExtracter::getPoints(*(geom[1]), pts1);
+
+	PointExtracter pe(pts0);
+	geom[0]->apply_ro(&pe);
+
+	PointExtracter pe1(pts1);
+	geom[1]->apply_ro(&pe1);
 
 #if GEOS_DEBUG
 	std::cerr << "PointExtracter found "
