--- FindMySQL.cmake_upstream	2011-02-24 18:17:37.064728000 +0100
+++ FindMySQL.cmake	2011-02-24 18:25:45.187647100 +0100
@@ -38,7 +38,7 @@
 endif(WIN32)
 
 if(WIN32)
-   string(TOLOWER ${CMAKE_BUILD_TYPE} CMAKE_BUILD_TYPE_TOLOWER)
+   string(TOLOWER "${CMAKE_BUILD_TYPE}" CMAKE_BUILD_TYPE_TOLOWER)
 
    # path suffix for debug/release mode
    # binary_dist: mysql binary distribution
@@ -77,6 +77,8 @@
 
 if(WIN32)
    set(MYSQL_LIB_PATHS
+      # added this path to find the library on Windows:
+      $ENV{MYSQL_DIR}/lib
       $ENV{MYSQL_DIR}/lib/opt
       $ENV{MYSQL_DIR}/client/release
       $ENV{ProgramFiles}/MySQL/*/lib/opt
@@ -96,7 +98,7 @@
       PATH_SUFFIXES
       mysql
    )
-   find_library(MYSQL_LIBRARIES NAMES mysqlclient
+   find_library(MYSQL_LIBRARIES NAMES mysqlclient_r
       PATHS
       ${MYSQL_LIB_PATHS}
    )
